baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/DangerousCollapseKeysUsage.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = SeverityLevel.ERROR,
+        summary = "Disallow usage of .collapseKeys() in EntryStream(s).")
+public final class DangerousCollapseKeysUsage extends BugChecker implements BugChecker.MethodInvocationTreeMatcher {
+    private static final long serialVersionUID = 1L;
+    private static final String ERROR_MESSAGE = "The collapseKeys API of EntryStream should be avoided. The "
```

Comment:
"should" implies info/warning, not ` SeverityLevel.ERROR`.

Reply:
Would you suggest changing the error level, or the language? I'm inclined to change this to `The collapseKeys API of EntryStream must be avoided`.

Reply:
I'd prefer setting this to WARNING to reduce friction of rollout. I might suggest adding a check for a sort called prior to this method, to avoid flagging cases that are safe.

Reply:
I've removed most of the internal usage so rollout should be friction free and we can keep it at ERROR. I've changed the `should` to `must`.